addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3969,16 +3969,29 @@ export class OdooEditor extends EventTarget {
             let previousSelection; // Is used to stop if `modify` doesn't move the selection.
             const hasSelectionChanged = (oldSelection = {}) => {
                 const newSelection = this.document.getSelection();
+                if (!newSelection.rangeCount) {
+                    return false;
+                }
+                const newRange = newSelection.getRangeAt(0);
                 return (
-                    oldSelection.anchorNode !== newSelection.anchorNode ||
-                    oldSelection.anchorOffset !== newSelection.anchorOffset ||
-                    oldSelection.focusNode !== newSelection.focusNode ||
-                    oldSelection.focusOffset !== newSelection.focusOffset
+                    oldSelection.anchorNode !== newRange.startContainer ||
+                    oldSelection.anchorOffset !== newRange.startOffset ||
+                    oldSelection.focusNode !== newRange.endContainer ||
+                    oldSelection.focusOffset !== newRange.endOffset
                 );
             };
             while (ZERO_WIDTH_CHARS.includes(adjacentCharacter) && hasSelectionChanged(previousSelection)) {
                 const selection = this.document.getSelection();
-                previousSelection = {...selection};
```

Comment:
No need of all these changes, it seems something wrong with spread syntax.

Reply:
Thanks for pointing this out @msh-odoo, upon debugging more on {...spreading} of an object i found that the `Selection` object is a DOM interface that represents the text selected by the user or the current position of the caret. It does not implement the iterable protocol, so methods like `for...of ` or spreading (...) won't work directly on it.
So i believe we need to keep some of those changes like manually assigning previousSelection.
```js
                previousSelection = {
                    anchorNode: selection.anchorNode,
                    anchorOffset: selection.anchorOffset,
                    focusNode: selection.focusNode,
                    focusOffset: selection.focusOffset,
                };
```
I will update the PR with necessary changes.